src/components/StudentsTable.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <AgGridVue
+    :columnDefs="studentsColumnDefs"
+    :defaultColDef="defaultStudentsColumnDef"
+    :rowData="studentsStore.students"
+    @gridReady="onGridReady"
+    class="students-table ag-theme-quartz-dark"
+  />
+</template>
+
+<script setup lang="ts">
+import { ref, shallowRef } from 'vue'
+import 'ag-grid-community/styles/ag-grid.css'
+import 'ag-grid-community/styles/ag-theme-quartz.css'
+import { AgGridVue } from 'ag-grid-vue3'
+import type { GridReadyEvent, GridApi } from 'ag-grid-community'
+
+import { useStudentsStore } from '@/stores/students'
+import {
+  getDefaultStudentsColDef,
+  getStudentsColDefs,
+  type StudentsTableRowData
+} from '@/helpers/columnDefinitions'
+import { createGridError } from '@/utils/createGridError'
+
+const studentsStore = useStudentsStore()
+
+const studentsColumnDefs = ref(getStudentsColDefs())
+const defaultStudentsColumnDef = ref(getDefaultStudentsColDef())
```

Comment:
do we need reactivity on this property? also why you decided to use function to create this object? I think default column def should be constant